app/Services/Fastly.php

```diff
@@ -2,52 +2,51 @@
 
 namespace Rogue\Services;
 
+use Rogue\Models\Post;
 use DoSomething\Gateway\Common\RestApiClient;
 
 class Fastly extends RestApiClient
 {
+    /**
+     * The Fastly service for this app.
+     */
+    protected $service;
+
     /**
      * Create a new Fastly API client.
      */
     public function __construct()
     {
-        $url = config('services.fastly.url');
+        $this->service = config('services.fastly.service_id');
 
-        $options = [
+        parent::__construct(config('services.fastly.url'), [
             'headers' => [
                 'Fastly-Key' => config('services.fastly.key'),
                 'Accept'     => 'application/json',
             ],
-        ];
+        ]);
+    }
 
-        parent::__construct($url, $options);
+    /**
+     * Purge any cached content for the given post.
+     */
+    public function purge(Post $post)
+    {
+        return $this->purgeKey('post-' . $post->id);
     }
 
     /**
      * Purge object from Fastly cache based on give cache key
      *
      * @param $cacheKey String
      */
-    public function purgeKey($cacheKey)
+    protected function purgeKey($cacheKey)
     {
-        $fastlyConfigured = config('features.glide') &&
-            ! is_null(config('services.fastly.url')) &&
-            ! is_null(config('services.fastly.key')) &&
-            ! is_null(config('services.fastly.service_id')) &&
-            isset($cacheKey);
-
-        if (! $fastlyConfigured) {
-            info('image_cache_purge_failed', ['response' => 'Fastly not configured correctly on this environment.']);
-
-            return null;
+        if (! $this->service) {
+            return;
         }
 
-        $service = config('services.fastly.service_id');
-
-        $purgeResponse = $this->post('service/'.$service.'/purge/'.$cacheKey);
-
+        $purgeResponse = $this->post('service/'.$this->service.'/purge/'.$cacheKey);
         info('image_cache_purge_successful', ['response' => $purgeResponse]);
-
-        return $purgeResponse;
```

Comment:
Do we not want to return anything here? The result here is also what's being returned by `purge()` right?

Reply:
Since we never use this API response in our app logic, I decided not to return it. I updated these to be `void` functions in 2d22da98 to make this a little bit more straightforward.